### PR TITLE
HTTPS urls

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
Lots of URLs were HTTP redirect to HTTPS. Changed those and checked them
if there's actual content. Inspired by #60004

They're mostly homepage URLs so there are probably a lot of source URLs left on HTTP which we should fix.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by rycee — APPROVED** [review]
Looks good to me. I tried a bunch of random links and they worked fine. Thanks!

**Comment by Infinisil** [comment]
For reference, previous efforts for moving to https (list might not be complete): #13809 #30906 #34836 #42665 #42676 #50989 #51001 #51329

**Review by c0bw3b — APPROVED** [review]

**Comment by JohnAZoidberg** [comment]
If found pretty much all of those because repology said they automatically redirect to HTTPS when you try to access HTTP: https://repology.org/repository/nix_unstable/problems

**Review by matthewbauer — APPROVED** [review]
Looks good to me. Moving fetchurl to https is not as important because we have hashes to ensure integrity, but it also doesn't hurt.

**Comment by Infinisil** [comment]
@matthiasbeyer It's important for whenever somebody updates the hashes

**Comment by matthiasbeyer** [comment]
@Infinisil you wanted @matthewbauer probably.

File: pkgs/applications/editors/eclipse/plugins.nix [hunk 1]
```diff
@@ -320,26 +320,26 @@ rec {
...
+        url = "https://www2.in.tum.de/projects/cup/eclipse/plugins/de.tum.in.www2.CupPlugin_${version}.jar";
         sha256 = "022phbrsny3gb8npb6sxyqqxacx138q5bd7dq3gqxh3kprx5chbl";
       })
     ];
 
     propagatedBuildInputs = [ zest ];
 
     meta = with stdenv.lib; {
-      homepage = http://www2.cs.tum.edu/projects/cup/eclipse.php;
+      homepage = https://www2.cs.tum.edu/projects/cup/eclipse.php;
```
Comment (markuskowa) [inline pkgs/applications/editors/eclipse/plugins.nix]: link is dead

File: pkgs/tools/text/unoconv/unoconv-python3.patch [hunk 2]
```diff
@@ -360,7 +360,7 @@ index 30e6706..f72cf08 100755
 -        print >>sys.stderr, "       http://github.com/dagwieers/unoconv/issues"
 +        print("unoconv: Cannot find a suitable office installation on your system.", file=sys.stderr)
 +        print("ERROR: Please locate your office installation and send your feedback to:", file=sys.stderr)
-+        print("       http://github.com/dagwieers/unoconv/issues", file=sys.stderr)
++        print("       https://github.com/dagwieers/unoconv/issues", file=sys.stderr)
```
Comment (markuskowa) [inline pkgs/tools/text/unoconv/unoconv-python3.patch]: That is the content of a patch. Seems not necessary to fix URL in the print statement.

Comment (JohnAZoidberg) [reply pkgs/tools/text/unoconv/unoconv-python3.patch]: I decided to keep it in anyways, doesn't hurt if it's part of the "addition" in the patch :D (And we all know that GitHub does HTTPS)

File: pkgs/applications/editors/eclipse/plugins.nix (hunk 1 above)
Comment (JohnAZoidberg) [reply pkgs/applications/editors/eclipse/plugins.nix]: hmmm, how did I miss that? Seems like they don't have support HTTPS but have different content there.

**Pull request merged**